src/+otp/+pendulum/PendulumProblem.m

```diff
@@ -0,0 +1,79 @@
+classdef PendulumProblem < otp.Problem
+    properties (SetAccess = private)
+        RhsMass
+    end
+    
+    methods
+        function obj = PendulumProblem(timeSpan, y0, parameters)
+            % Constructs a problem
+            obj@otp.Problem('Pendulum', [], timeSpan, y0, parameters);
+        end
+        
+        function [x, y] = convert2Cartesian(obj, y, includeOrigin)
+            angles = y(1:end/2, :);
+            lengths = obj.Parameters.lengths(:);
+            x = cumsum(lengths .* sin(angles), 1);
+            y = cumsum(-lengths .* cos(angles), 1);
+            
+            if nargin > 2 && includeOrigin
+                z = zeros(1, size(y, 2));
+                x = [z; x];
+                y = [z; y];
+            end
+        end
+    end
+    
+    methods (Access = protected)
+        function onSettingsChanged(obj)
+            g = obj.Parameters.g;
+            lengths = obj.Parameters.lengths(:);
+            masses = obj.Parameters.masses(:);
+            
+            numBobs = min(numel(lengths), numel(masses));
+            lengths = lengths(1:numBobs);
+            masses = masses(1:numBobs);
+            cumulativeMasses = cumsum(masses, 'reverse');
+            
+            offDiagScaling = -1 ./ (lengths(1:end-1) .* lengths(2:end) .* masses(1:end-1));
+            cDiag = [1 / (lengths(1)^2 * masses(1)); (masses(1:end-1) + masses(2:end)) ./ (lengths(2:end).^2 .* masses(1:end-1) .* masses(2:end))];
+            
+            obj.Rhs = otp.Rhs(@(t, y) otp.pendulum.f(t, y, lengths, cumulativeMasses, g, offDiagScaling, cDiag));
```

Comment:
JVP? If it's too difficult, then meh.

Reply:
Jacobian and JVP is complicated.  We can add it later if needed.